<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>KMeansDialog</class>
 <widget class="QDialog" name="KMeansDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>483</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <widget class="QGroupBox" name="pointsGroupBox">
     <property name="title">
      <string>Points</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QGroupBox" name="minGroupBox">
        <property name="title">
         <string/>
        </property>
        <layout class="QFormLayout" name="formLayout_4">
         <item row="0" column="0">
          <widget class="QLabel" name="minLabel">
           <property name="text">
            <string>Min</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QDoubleSpinBox" name="minDoubleSpinBox"/>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="maxGroupBox">
        <property name="title">
         <string/>
        </property>
        <layout class="QFormLayout" name="formLayout_5">
         <item row="0" column="0">
          <widget class="QLabel" name="maxLabel">
           <property name="text">
            <string>Max</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QDoubleSpinBox" name="maxDoubleSpinBox">
           <property name="decimals">
            <number>2</number>
           </property>
           <property name="maximum">
            <double>200.000000000000000</double>
           </property>
           <property name="value">
            <double>100.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="NGroupBox">
        <property name="title">
         <string/>
        </property>
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="NLabel">
           <property name="text">
            <string>N</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="NSpinBox">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>20</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="generatePointsButton">
        <property name="text">
         <string>Generate Points</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="kmeansGroupBox">
     <property name="title">
      <string>K - Means</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QGroupBox" name="kGroupBox">
        <property name="title">
         <string/>
        </property>
        <layout class="QFormLayout" name="formLayout_2">
         <item row="0" column="0">
          <widget class="QLabel" name="kLabel">
           <property name="text">
            <string>k</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="kSpinBox">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="value">
            <number>3</number>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="iterSpinBox">
           <property name="value">
            <number>2</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="iterLabel">
           <property name="text">
            <string>Iter</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="metricGroupBox">
        <property name="title">
         <string>Metric</string>
        </property>
        <layout class="QFormLayout" name="formLayout_6">
         <item row="0" column="0" colspan="2">
          <widget class="QComboBox" name="metricComboBox">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="currentText">
            <string>Euclidean (L2)</string>
           </property>
           <property name="maxCount">
            <number>10</number>
           </property>
           <item>
            <property name="text">
             <string>Euclidean (L2)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Manhattan (L1)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Hamming</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="clusterButton">
        <property name="text">
         <string>Cluster</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>KMeansDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>KMeansDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>generatePointsButton</sender>
   <signal>clicked()</signal>
   <receiver>KMeansDialog</receiver>
   <slot>applyGeneratePoints()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>73</x>
     <y>267</y>
    </hint>
    <hint type="destinationlabel">
     <x>238</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clusterButton</sender>
   <signal>clicked()</signal>
   <receiver>KMeansDialog</receiver>
   <slot>applyClustering()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>219</x>
     <y>267</y>
    </hint>
    <hint type="destinationlabel">
     <x>238</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>clicked()</signal>
  <slot>applyGeneratePoints()</slot>
  <slot>applyClustering()</slot>
 </slots>
</ui>
